components/kyma-environment-broker/internal/process/upgrade_cluster/initialisation.go

```diff
@@ -115,20 +115,6 @@ func (s *InitialisationStep) Run(operation internal.UpgradeClusterOperation, log
 }
 
 func (s *InitialisationStep) initializeUpgradeShootRequest(operation internal.UpgradeClusterOperation, log logrus.FieldLogger) (internal.UpgradeClusterOperation, time.Duration, error) {
-	// rewrite necessary data from ProvisioningOperation to operation internal.UpgradeOperation
-	provisioningOperation, err := s.operationStorage.GetProvisioningOperationByInstanceID(operation.InstanceID)
-	if err != nil {
-		log.Errorf("while getting provisioning operation from storage")
-		return operation, s.timeSchedule.Retry, nil
-	}
-
-	operation, delay := s.operationManager.UpdateOperation(operation, func(op *internal.UpgradeClusterOperation) {
-		op.ProvisioningParameters = provisioningOperation.ProvisioningParameters
-	}, log)
-	if delay != 0 {
-		return operation, delay, nil
-	}
```

Comment:
I am not familiar with the logic to a sufficient extent, but I was wondering if we should get `ProvisioningParameters` from the last successful operation? In KEB, certain params are expected to be populated but sometimes these are added in the broker API endpoint, sometimes in the initialization step so not sure which workflow would fit best here.

Do we know if the rest of the upgrade queue expects these to be present?

Reply:
maybe @piotrmiskiewicz remembers?

Reply:
It is being done actually, when the upgrade cluster operation is created, the ProvisioningParameters is copies over from the Instance, which holds the latest state of the params:
https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/orchestration/manager/upgrade_cluster.go#L60